core/src/main/java/discord4j/core/shard/GatewayBootstrap.java

```diff
@@ -720,6 +734,13 @@ private StoreService initStoreService() {
                 .apply(storeService);
     }
 
+    private EntityRetrievalStrategy initEntityRetrievalStrategy() {
+        if (entityRetrievalStrategy != null) {
+            return entityRetrievalStrategy;
+        }
+        return EntityRetrievalStrategy.STORE;
```

Comment:
LGTM overall. We just need to agree which default setting to use.

Reply:
Yeah I've put this for now since you deprecated the old methods that did the fallback to REST, so I assumed you were against that behavior being the default.
But on another hand it would be less work for people migrating from 3.0.x if we kept it. With STORE strategy they would always get empty monos instead of ClientException, so they would need to migrate their error handlers. So yeah both options are to consider and discuss.

Reply:
I want to avoid changing a 3.0 behavior on a method that would keep its signature so would prefer having the "mixed" strategy as default.

Reply:
I agree that makes sense, I'm gonna change it tomorrow then. Unless you want input from others before I change anything?

Reply:
Change it for now, but we expect feedback from @darichey . and @danthonywalker OK'd the PR

Reply:
Resolved in 3cdbf0e